src/main/java/pl/agh/cs/io/WindowsPerExe.java

```diff
@@ -0,0 +1,43 @@
+package pl.agh.cs.io;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class WindowsPerExe {
+    private String exePath;
+    private Set<Integer> processIds;
+
+    public WindowsPerExe(String exePath, Set<Integer> processIds) {
+        this.exePath = exePath;
+        this.processIds = processIds;
+    }
+
+    public static WindowsPerExe fromWindow(Window window) {
```

Comment:
nie lepiej zrobić konstruktor, który przyjmuje `Window`?

Reply:
> checkstyle jest czerwony: https://travis-ci.org/kswierad/TimeGuard/builds/523439075?utm_source=github_status&utm_medium=notification

Poprawiłem

> nie lepiej zrobić konstruktor, który przyjmuje Window?

To trochę taki idiom Javowy - metoda fabryczna o czytelnej nazwie zamiast dziwnych konstruktorów

Reply:
ale wtedy chyba konstruktor powinien być private